apps/hash-frontend/src/components/hooks/use-account-pages.ts

```diff
@@ -18,13 +19,23 @@ export const useAccountPages = (
   ownedById: OwnedById,
   includeArchived?: boolean,
 ): AccountPagesInfo => {
-  const { data, loading } = useQuery<
+  const [getAccountPages, { data, loading }] = useLazyQuery<
     GetAccountPagesTreeQuery,
     GetAccountPagesTreeQueryVariables
   >(getAccountPagesTree, {
     variables: { ownedById, includeArchived },
   });
 
+  const { hashInstance } = useHashInstance();
+
+  useEffect(() => {
+    if (!hashInstance?.properties.pagesAreEnabled) {
+      return;
+    }
+
+    void getAccountPages();
+  }, [getAccountPages, hashInstance?.properties.pagesAreEnabled]);
+
```

Comment:
I think you could also just use the `skip` parameter of `useQuery` to simplify this a bit. I think it's something like:
```TypeScript
  const { data, loading } = useQuery<
    GetAccountPagesTreeQuery,
    GetAccountPagesTreeQueryVariables
  >(getAccountPagesTree, {
    variables: { ownedById, includeArchived },
    skip: !hashInstance?.properties.pagesAreEnabled
  });
```

Reply:
Ahh much simpler, thank you – https://github.com/hashintel/hash/pull/3094/commits/bcf9fd912379c6497e55bb4c62e889ef6b7db4a5